qiita_pet/static/js/networkVue.js

```diff
@@ -0,0 +1,769 @@
+          }
+          // Format node list data
+          for(var i = 0; i < data.nodes.length; i++) {
+            vm.nodes.push({id: data.nodes[i][2], label: data.nodes[i][3], type: data.nodes[i][1], group: data.nodes[i][0]});
+          }
+          vm.drawProcessingGraph('processing-results');
+
+          // At this point we can show the graph and hide the job list
+          $("#processing-network-div").show();
+          $("#processing-job-div").hide();
+        }
+      })
+        .fail(function(object, status, error_msg) {
+          // Show an error message if something wrong happen, rather than
+          // leaving the spinning wheel of death in there.
+          $("#processing-network-div").html("Error loading graph: " + status + " " + error_msg);
+          $("#processing-network-div").show();
+          $("#processing-job-div").hide();
+        }
+      );
+    },
+
+    /**
+     *
+     * This function check the status of the jobs that generates the initial
+     * set of files.
+     *
+     **/
+    checkInitialJobs: function () {
+      let vm = this;
+      $.get(vm.portal + vm.jobsEndpoint, function(data) {
+        $("#processing-job-div").html("");
+        $("#processing-job-div").append("<p>Hang tight, we are generating the initial set of files: </p>");
+        for(var jobid in data){
+          var contents = "<b> Job: " + jobid + "</b> Status: " + data[jobid]['status'];
+          // Only show step if error if they actually have a useful message
+          if (data[jobid]['step']) {
+            contents = contents + " Step: " + data[jobid]['step'] + "</br>";
+          }
+          if (data[jobid]['error']) {
+            contents = contents + " Error: " + data[jobid]['error'] + "</br>";
+          }
+          $("#processing-job-div").append(contents);
+        }
+      })
+        .fail(function(object, status, error_msg) {
+          $("#processing-job-div").html("Error loading job information: " + status + " " + error_msg);
+        }
+      );
+    }
+  },
+
+  /**
+   *
+   * This function gets called by Vue once the HTML template is ready in the
+   * actual DOM. We can use it as an "init" function.
+   *
+   **/
+  mounted() {
+    let vm = this;
+    vm.nodes = [];
+    vm.edges = [];
+    vm.runningJobs = [];
+    vm.inConstructionJobs = 0;
+    vm.workflowId = null;
+    vm.colorScheme = {
+      'success': '#00cc00',
+      'running': '#cccc00',
+      'error': '#ff3333',
+      'in_construction': '#BBBBBB'};
+    show_loading('processing-network-div');
+    $("#processing-network-div").hide();
+
+    $('#run-btn').on('click', function() { vm.runWorkflow(); });
+    $('#run-btn-div').hide();
+    // This call to udpate graph will take care of updating the jobs
+    // if the graph is not available
+    vm.updateGraph();
+    setInterval(function() {
+      // Only update if the graph has not been generated yet
+      if (vm.nodes.length == 0) {
```

Comment:
This is the interval the graph is gonna be refreshed, currently 5 seconds ... to fast/slow? should it be smarter? like only refresh the GUI if there are actual changes in the network?